django_toolbox/shopify_graphql.py

```diff
@@ -62,7 +80,17 @@ def run_query(*args, **kwargs):
     retry_count = 0
     while True:
         try:
-            return _run_query(*args, **kwargs)
+            with elasticapm.capture_span("GraphQL Query", span_type="graphql") as span:
```

Comment:
I think we should reorganize how we're saving the data. According to the https://github.com/elastic/apm-agent-python/blob/master/elasticapm/instrumentation/packages/requests.py example. We should use dynamic portion as name. I would go for `meta_data["query_id"]` instead of `"GraphQL Query"`.

Reply:
The problem with meta_data['query_id'] is that it's not available before the graphql response is retrieved. However this works, but its kind of ugly:

```
            with elasticapm.capture_span("GraphQL Query", span_type="graphql") as span:
                content = _run_query(*args, **kwargs)
                meta_data = _get_query_meta_data(content)
                if meta_data:
                    span.name = meta_data['query_id']
                    span.labels = {
                        "query_id": meta_data["query_id"],
                        "requested_query_cost": meta_data["requested_query_cost"],
                        "actual_query_cost": meta_data["actual_query_cost"],
                    }
```
Alternatively, we can do something like this:

```
with elasticapm.capture_span(kwargs['query'][:50] + "...", span_type="graphql") as span:
```

In both cases , I would add the query_id into labels as well so we can use it for filtering.


WDYT?

Reply:
Ah yeah :( I would vouch for parsing the query and extracting the name or first entity that's being queried, but let's do that later. We could also add optional name parameter and name the span manually.